docs/en/configuration.md

```diff
@@ -231,9 +231,31 @@ visible locales for this object.
 Note: Although these objects will be filtered in the front end, this filter is disabled
 in the CMS in order to allow access by site administrators in all locales.
 
-## Locale detection
+## Routing and Locale Detection
 
-When a visitor lands on the home page for the first time, Fluent can attempt to detect that user's locale based
+The `DetectLocaleMiddleware` will detect if a locale has been requested (or is default) and is not the current
+locale, and will redirect the user to that locale if needed.
+
+Will cascade through different checks in order:
+1. Routing path (e.g. `/de/ueber-uns`)
+2. Request variable (e.g. `ueber-uns?FluentLocale=de`)
+3. Domain (e.g. `http://example.de/ueber-uns`)
+4. Session (if a session is already started)
+5. Cookie (if `DetectLocaleMiddleware.persist_cookie` is configured)
+6. Request headers (if `FluentDirectorExtension.detect_locale` is configured)
+
+Additionally, detected locales will be set in cookies.
+This behaviour can be configured through `DetectLocaleMiddleware.persist_cookie`.
+To solely rely on sessions and stateless request data (routing path, request
+variable or domain), configure as follows:
```

Comment:
@robbieaverill Technically, this is no longer accurate advice. To *store* in sessions, you have to force a session to be started *before* this logic kicks in. Either in the same request (through higher priority middleware), or more commonly through an explicit `Session::start()` in your own logic. It's a bit weird to rely on something outside of this module logic flow to trigger this behaviour, and it can lead to side effects. If you naively rely on sessions to store language, you might end up diagnosing issues where sometimes sessions aren't started, and language selections "don't stick". Can you expand on that in the docs?

Reply:
Cool, I've pushed an updated commit - I hope it represents what you mean 😃